ClosedXML/Excel/Cells/XLCellsCollection.cs

```diff
@@ -212,22 +212,28 @@ public int FirstColumnUsed(int rowStart, int columnStart, int rowEnd, int column
         {
             int finalRow = rowEnd > MaxRowUsed ? MaxRowUsed : rowEnd;
             int finalColumn = columnEnd > MaxColumnUsed ? MaxColumnUsed : columnEnd;
+            int firstColumnUsed = finalColumn;
+            var found = false;
             for (int ro = rowStart; ro <= finalRow; ro++)
             {
                 if (rowsCollection.TryGetValue(ro, out Dictionary<Int32, XLCell> columnsCollection))
                 {
-                    for (int co = columnStart; co <= finalColumn; co++)
+                    for (int co = columnStart; co <= firstColumnUsed; co++)
                     {
                         if (columnsCollection.TryGetValue(co, out XLCell cell)
                             && !cell.IsEmpty(includeFormats)
-                            && (predicate == null || predicate(cell)))
-
-                            return co;
+                            && (predicate == null || predicate(cell))
+                            && co <= firstColumnUsed)
+                        {
+                            firstColumnUsed = co;
+                            found = true;
+                            break;
```

Comment:
you can optimize this for most use cases - big speed improvement for files with many rows:
```
if (co == columnStart) return co;
break;
```